pkg/agent/proxy/proxier.go

```diff
@@ -513,21 +496,35 @@ func (p *proxier) installServices() {
 				}
 			}
 
-			updateInstalledEndpointsMap := func(endpoint k8sproxy.Endpoint) {
-				// If the Endpoint is newly installed, add a reference.
-				if _, ok := endpointsInstalled[endpoint.String()]; !ok {
+			if mcsLocalService != nil {
+				allReachableEndpoints = append(allReachableEndpoints, mcsLocalService.Endpoint)
+			}
+
+			updatedEndpointsInstalled := map[string]k8sproxy.Endpoint{}
+			for _, endpoint := range allReachableEndpoints {
+				updatedEndpointsInstalled[endpoint.String()] = endpoint
+				// If the Endpoint is not newly installed, remove it from stale endpoints.
```

Comment:
"remove it from stale endpoints": that doesn't sound correct. You are removing it from `endpointsInstalled`, which doesn't sound like a list of stale endpoints